modules/collabora_online_group/tests/src/Kernel/PermissionTest.php

```diff
@@ -0,0 +1,113 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Drupal\Tests\collabora_online_group\Kernel;
+
+use Drupal\Tests\group\Kernel\GroupKernelTestBase;
+use Drupal\Tests\media\Traits\MediaTypeCreationTrait;
+
+/**
+ * Tests group permissions.
+ */
+class PermissionTest extends GroupKernelTestBase {
+
+    use MediaTypeCreationTrait;
+
+    /**
+     * {@inheritdoc}
+     */
+    protected static $modules = [
+        'file',
+        'media',
+        'group',
+        'groupmedia',
+    ];
+
+    /**
+     * Tests that group permissions are properly created.
+     */
+    public function testGroupPermissions(): void {
+        // Generate types: groups and medias.
+        $group_type_1 = $this->createGroupType(['id' => 'group_1']);
+        $group_type_2 = $this->createGroupType(['id' => 'group_2']);
+        $group_type_3 = $this->createGroupType(['id' => 'group_3']);
+        $media_type_1 = $this->createMediaType('file', ['id' => 'media_1']);
+        $media_type_2 = $this->createMediaType('file', ['id' => 'media_2']);
+
+        // Enable relation plugins in groups.
+        $this->entityTypeManager()->getStorage('group_relationship_type')
+            ->createFromPlugin($group_type_1, 'group_media:' . $media_type_1->id(), [
+                'group_cardinality' => 0,
+                'entity_cardinality' => 1,
+                'use_creation_wizard' => FALSE,
+            ])
+            ->save();
+        $this->entityTypeManager()->getStorage('group_relationship_type')
+            ->createFromPlugin($group_type_2, 'group_media:' . $media_type_1->id(), [
+                'group_cardinality' => 0,
+                'entity_cardinality' => 1,
+                'use_creation_wizard' => FALSE,
+            ])
+            ->save();
+        $this->entityTypeManager()->getStorage('group_relationship_type')
+            ->createFromPlugin($group_type_2, 'group_media:' . $media_type_2->id(), [
+                'group_cardinality' => 0,
+                'entity_cardinality' => 1,
+                'use_creation_wizard' => FALSE,
+            ])
+            ->save();
+
+        // Check that permissions are generated for the groups.
+        // Save current permissions count.
+        $permission_handler = \Drupal::service('group.permissions');
+        $count_group_1 = count($permission_handler->getPermissionsByGroupType($group_type_1));
+        $count_group_2 = count($permission_handler->getPermissionsByGroupType($group_type_2));
+        $count_group_3 = count($permission_handler->getPermissionsByGroupType($group_type_3));
```

Comment:
Aren't these counts always the same?
If so, we can just assert the count instead of introducing a variable.

Reply:
I see now there is a lot of them :)
31, 45, 17.
So we could expect the numbers to change with different versions of group module.

Reply:
Added some changes with this suggestion, I think is clearer and easier to read now.